### PR TITLE
Allow a user to pass in None for chunk dimensions

### PR DESCRIPTION
**Pull request by cpcloud**
Example:

Instead of explicitly listing dimensions, like this:
```python
In [1]: import dask.array as da

In [2]: x = np.random.randn(1464, 73, 144)

In [3]: d = da.from_array(x, chunks=(500, 73, 144))
```

You can let dask figure it out for you:

```python
In [4]: d = da.from_array(x, chunks=(500, None, None))

In [5]: d.chunks
Out[5]: ((500, 500, 464), (73,), (144,))
```

### CODE REVIEW

File: dask/array/tests/test_rechunk.py [hunk 1]
```diff
@@ -164,3 +164,9 @@ def test_rechunk_with_empty_input():
     x = da.ones((24, 24), chunks=(4, 8))
     assert x.rechunk(chunks={}).chunks == x.chunks
     assert raises(ValueError, lambda: x.rechunk(chunks=()))
+
+
+def test_rechunk_with_null_dimensions():
+    x = da.ones((24, 24), chunks=(4, 8))
+    assert (x.rechunk(chunks=(None, 4)).chunks ==
+            da.ones((24, 24), chunks=(24, 4)).chunks)
```
Comment (mrocklin) [inline dask/array/tests/test_rechunk.py]: Hrm, I would have interpreted this as "don't change the current chunking of the first dimension"

**Comment by mrocklin** [comment]
@shoyer do you have thoughts here?  You have a good eye for API.

File: dask/array/tests/test_rechunk.py (hunk 1 above)
Comment (cpcloud) [inline dask/array/tests/test_rechunk.py]: yes, me too when i was writing it. i think that'd be more intuitive

**Comment by shoyer** [comment]
Yep, I think this is a great API. I already did essentially the exact same thing for xray's dask API, except that because you can use a dictionary, you can signal that chunking should use the default size by simply omitting a key instead of explicitly using `None`.

**Comment by mrocklin** [comment]
This looks good to me.  Looks like it needs a rebase.

**Comment by mrocklin** [comment]
or merge with master

**Comment by jcrist** [comment]
@cpcloud : can you fix the merge conflicts so we can merge this?

**Comment by cpcloud** [comment]
yep on it, sorry for the delay ... give me about 20 minutes

**Comment by cpcloud** [comment]
Resolved. Should I merge on passing?

**Comment by jcrist** [comment]
Looks good to me. I'd say yes.

**Comment by mrocklin** [comment]
+1

**Pull request merged**